Foundation/NSDictionary.swift

```diff
@@ -156,13 +156,44 @@ public class NSDictionary : NSObject, NSCopying, NSMutableCopying, NSSecureCodin
     }
     
     public func copyWithZone(zone: NSZone) -> AnyObject {
-        NSUnimplemented()
+        if self.dynamicType === NSDictionary.self {
+            // NSDictionary is immutable; just return ourself
+            return self
+        } else if self.dynamicType === NSMutableDictionary.self {
+            // Otherwise, create a new NSDictionary object
+            
+            // TODO: speed up?
```

Comment:
technically you could create a NSDictionary by returning one created via the storage directly.